libraries/google/compute/property/instancegroupmanager_current_actions.rb

```diff
@@ -0,0 +1,50 @@
+# frozen_string_literal: false
+
+# ----------------------------------------------------------------------------
+#
+#     ***     AUTO GENERATED CODE    ***    AUTO GENERATED CODE     ***
+#
+# ----------------------------------------------------------------------------
+#
+#     This file is automatically generated by Magic Modules and manual
+#     changes will be clobbered when the file is regenerated.
+#
+#     Please read more about how to change this file in README.md and
+#     CONTRIBUTING.md located at the root of this package.
+#
+# ----------------------------------------------------------------------------
+module GoogleInSpec
+  module Compute
+    module Property
+      class InstanceGroupManagerCurrentactions
+        attr_reader :abandoning
+
+        attr_reader :creating
+
+        attr_reader :creating_without_retries
+
+        attr_reader :deleting
+
+        attr_reader :none
+
+        attr_reader :recreating
+
+        attr_reader :refreshing
+
+        attr_reader :restarting
+
+        def initialize(args = nil)
+          return if args.nil?
```

Comment:
Since args is a Hash, you could do all of these assignments as a one-liner if you wanted to.

Something like 'args.each |x| instance_variable_set("@#{x}", args[x])' (pretty sure there's an even cleaner way).

More of a heads-up. I don't have an opinion either way.

Reply:
I believe it is possible for these nested objects to contain further nested objects, which require passing args[x] to an initialize method. So I think it's better to keep these similar to the general resource parse method.

Reply:
Good call.